pramen/core/src/main/scala/za/co/absa/pramen/core/sql/SqlGeneratorBase.scala

```diff
@@ -49,66 +68,95 @@ abstract class SqlGeneratorBase(sqlConfig: SqlConfig) extends SqlGenerator {
     if (columns.isEmpty) {
       "*"
     } else {
-      columns.map(col => escapeIdentifier(col)) .mkString(", ")
+      columns.map(col => escape(col)) .mkString(", ")
+    }
+  }
+
+  final def splitComplexIdentifier(identifier: String): Seq[String] = {
+    val trimmedIdentifier = identifier.trim
+
+    if (trimmedIdentifier.isEmpty) {
+      throw new IllegalArgumentException(f"Found an empty table name or column name ('$identifier').")
     }
+
+    val (escapeBegin, escapeEnd) = beginEndEscapeChars
+    val sameEscapeChar = escapeBegin == escapeEnd
+
+    val output = new ListBuffer[String]
+    val curColumn = new StringBuffer()
+    val len = trimmedIdentifier.length
+    var nestingLevel = 0
+    var i = 0
+
+    while (i < len) {
+      val c = trimmedIdentifier(i)
+      val previousChar = if (i == 0) ' ' else trimmedIdentifier(i - 1)
+      val nextChar = if (i == len - 1) ' ' else trimmedIdentifier(i + 1)
+
+      if (nestingLevel == 0 && c == '.') {
+        output += curColumn.toString
+        curColumn.setLength(0)
+      } else {
+        curColumn.append(c)
+      }
+
+      if (sameEscapeChar) {
+        if (c == escapeBegin) {
+          if (nestingLevel == 0) {
+            if (curColumn.length() != 1 && previousChar != escapeBegin && nextChar != escapeBegin)
```

Comment:
I'm thinking just not supporting `System "User"` fields for databases which use `"` as a quoting character for now. 😆  
It looks like a shortcut yes, but I don't want to overengineer too much before any use encounters such a case.  Otherwise, we need to consider escaping rules of different dbs, and proper handling of `\` in some cases.

But if we do encounter it, we can implement later.